ec2/spark_ec2.py

```diff
@@ -423,13 +426,14 @@ def get_spark_ami(opts):
         b=opts.spark_ec2_git_branch)
 
     ami_path = "%s/%s/%s" % (ami_prefix, opts.region, instance_type)
+    reader = codecs.getreader("ascii")
     try:
-        ami = urlopen(ami_path).read().strip()
-        print("Spark AMI: " + ami)
+        ami = reader(urlopen(ami_path)).read().strip()
```

Comment:
I was totally wrong.  Streamreader only deals with bytes, but because urlopen() is pulled from urllib.request in py3 and from urllib2 in py2, both produce bytestreams, which is why both work.

I can swap that call out for decode if it seems more obvious/maintainable.  The behavior will be equivalent.